importers/adultfriendfinder.go

```diff
@@ -0,0 +1,288 @@
+
+	bulk := c.Bulk()
+	bulk.Unordered()
+
+	for text := range threader {
+		if bc > 10000 {
+			bulk.Run()
+			if bar != nil {
+				bar.Add(bc)
+			}
+			bc = 0
+			bulk = c.Bulk()
+			bulk.Unordered()
+		}
+		// Split out the INSERT statement so we have the tuples we need
+		// Start by removing the first and last characters
+		text2 := text[31 : len(text)-2]
+
+		// Split on "),("
+		users := strings.Split(text2, "),(")
+
+		// Now loop through and process each line and INSERT as required
+		for _, row := range users {
+			strippedRow := strings.ReplaceAll(row, "'", "")
+			data := strings.Split(strippedRow, ",")
+
+			// format: pwsid, handle, email, lastvisit, password, send_email, show_lang.....
+			if len(data) < 4 {
+				if bar != nil {
+					bar.Increment()
+				}
+				// problem! error
+				fmt.Println("Invalid data, epxected a len > 4, got ", len(data), strippedRow)
+				continue
+			}
+
+			// We remove the underscore and convert it to member ID
+			memberid, err := strconv.Atoi(strings.Replace(data[0], "_", "", 1))
+			if err != nil {
+				fmt.Println("Invalid data read, conversion failed", data)
+				continue
+			}
+
+			// There may be multiple emails in the email column, get the index of the last email
+			// The default index of email is 2
+			lastEmailIndex := 2
+			for i := 2; i < len(data); i++ {
+				// extremely rough check if it's a valid email
+				if strings.Contains(data[i], "@") {
+					lastEmailIndex = i
+				} else {
+					break
+				}
+			}
+			// We only care about users with password hash
+			if len(data[lastEmailIndex+2]) == 40 {
+				// Import all the emails
+				for i := 2; i <= lastEmailIndex; i++ {
+					// Parse relevant entries
+					entry := AdultFriendFinderData{
+						MemberID:     memberid,
+						Email:        data[i],
+						Liame:        Reverse(data[i]),
+						PasswordHash: data[4],
+						Username:     data[1],
+						Breach:       "AdultFriendFinder2016",
+					}
+					bulk.Insert(entry)
+					bc += 1
+				}
+			} else {
+				// Here we insert those emails with commas
+				if len(data) > 4 {
+					hasPwd := false
+					for i := 4; i < len(data); i++ {
+						if len(data[i]) == 40 {
+							hasPwd = true
+						}
+					}
+					// Check if we skipped over some password hash
+					// This may happen if we have comma in the email
+					if hasPwd {
+						// fmt.Println("Invalid data read, password hash length does not match", row)
+						tmpData := strings.Split(row, "','")
+						// insert this entry
+						entry := AdultFriendFinderData{
+							MemberID:     memberid,
+							Email:        tmpData[2],
+							Liame:        Reverse(tmpData[2]),
+							PasswordHash: tmpData[4],
+							Username:     tmpData[1],
+							Breach:       "AdultFriendFinder2016",
+						}
+						bulk.Insert(entry)
+						bc += 1
+					} else {
+						if bar != nil {
+							bar.Increment()
+						}
+					}
+				} else {
+					if bar != nil {
+						bar.Increment()
+					}
+				}
+			}
+		}
```

Comment:
Instead of having all of the `bar.Increment()`'s and `bar.Add()`'s, it would be much tidier to just have a `bar.Add(len(users))` after the for loop instead (this should be equivalent)